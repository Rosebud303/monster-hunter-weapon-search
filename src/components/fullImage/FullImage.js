import React from 'react';
import { Link } from 'react-router-dom';
import './FullImage.scss';

const FullImage = ({picture, id, name}) => {
  return (
    <div>
      <div className='full-image'>
        <h2 className='image-name'>{name}</h2>
        <img className='full-picture' src={picture}/>
        <Link className='image-back' to={`/weapons/${id}`}>Back</Link>
      </div>
    </div>
  )
}

export default FullImage
