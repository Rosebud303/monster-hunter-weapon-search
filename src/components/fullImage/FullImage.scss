.full-image {
  width: 30%;
  margin: auto;
  background: linear-gradient( rgba(0, 0, 0, 0.7), rgba(0, 0, 0, 0.5) ), url('https://i.imgur.com/G7MEwRL.jpg');
  background-repeat: no-repeat;
  opacity: .8;
  background-position: center;
  padding: 20px;
  display: flex;
  flex-direction: column;
  align-items: center;
  box-shadow: 8px 8px 14px 1px #0c0a0a;
  border: 2px solid #11fadb;
  border-radius: 7px;
}

.image-back {
  color: white;
  text-decoration: none;
  font-size: 30px;
  font-weight: 800;
  &:hover {
    transition: .5s;
    color: red;
  }
}

.image-name {
  color: white;
}

.full-picture {
  width: 100%
}