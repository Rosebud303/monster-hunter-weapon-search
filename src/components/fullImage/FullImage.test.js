import React from 'react';
import { shallow } from 'enzyme';
import FullImage from './FullImage'



describe('', () => {
  let wrapper;
  let mockId = 1
  let mockPicture = 'picture.png'
  let mockName = 'Sword'
  let mockType = 'small-sword'
  let mockMin = 80
  let mockMax = 200
  let mockDmgType = 'fire'
  let mockRare = 1
  let mockCraft = false
  beforeEach(() => {
    wrapper = shallow (
      <FullImage id={mockId}
      name={mockName}
      picture={mockPicture}
      type={mockType}
      minDMG={mockMin}
      maxDMG={mockMax}
      damageType={mockDmgType}
      rarity={mockRare}
      craftable={mockCraft}/>
    )
  });

  it('should match snap shot', () => {
    expect(wrapper).toMatchSnapshot()
  });

});
