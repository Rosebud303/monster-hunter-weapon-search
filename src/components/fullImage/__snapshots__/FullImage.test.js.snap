// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snap shot 1`] = `
<div>
  <div
    className="full-image"
  >
    <h2
      className="image-name"
    >
      Sword
    </h2>
    <img
      className="full-picture"
      src="picture.png"
    />
    <Link
      className="image-back"
      to="/weapons/1"
    >
      Back
    </Link>
  </div>
</div>
`;
